step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/AutomationPackageResourceManager.java

```diff
@@ -66,6 +70,41 @@ public AutomationPackageResourceManager(ResourceManager resourceManager,
         this.mavenConfig = mavenConfig;
     }
 
+    /**
+     * This method is used to check that we could reuse and update the existing resource, to be called before updating the package resource
+     * @param apLibProvider the apLibProvider
+     * @param parameters the updated parameters
+     * @param allowToReuseOldResource the flag allowing or not to reuse libraries
+     */
+    public void validateUploadOrReuseAutomationPackageLibrary(AutomationPackageLibraryProvider apLibProvider,
+                                                          AutomationPackageUpdateParameter parameters, boolean allowToReuseOldResource) {
+        try {
+            File apLibrary = apLibProvider.getAutomationPackageLibrary();
+            if (apLibrary != null) {
+                // we can reuse the existing old resource in case it is identifiable (can be found by origin) and unmodifiable
+                List<Resource> oldResources = null;
+                if (apLibProvider.canLookupResources()) {
+                    oldResources = apLibProvider.lookupExistingResources(resourceManager, parameters.objectPredicate);
+                    if (oldResources != null && !oldResources.isEmpty()) {
+                        Resource oldResource = oldResources.get(0);
```

Comment:
You have a list of resources, but you're explicitly only checking the first. Is this intentional? If so, add a comment why.